spec/components/teaching_events/event_component_spec.rb

```diff
@@ -0,0 +1,193 @@
+require "rails_helper"
+
+describe TeachingEvents::EventComponent, type: "component" do
+  let(:event) { build(:event_api) }
+
+  describe "methods" do
+    let(:subject) { described_class.new(event: event) }
+
+    describe "#train_to_teach?" do
+      context "when the event is a TTT event" do
+        let(:event) { build(:event_api) }
+
+        it { expect(subject.train_to_teach?).to be(true) }
+      end
+
+      context "when the event is a question time event" do
+        let(:event) { build(:event_api, :question_time_event) }
+
+        it { expect(subject.train_to_teach?).to be(true) }
+      end
+
+      context "when the event is not a TTT event" do
+        let(:event) { build(:event_api, :school_or_university_event) }
+
+        it { expect(subject.train_to_teach?).to be(false) }
+      end
+    end
+
+    describe "#school_and_university?" do
+      context "when the event is a school and university event" do
+        let(:event) { build(:event_api, :school_or_university_event) }
+
+        it { expect(subject.school_and_university?).to be(true) }
+      end
+
+      context "when the event is a train to teach event" do
+        let(:event) { build(:event_api) }
+
+        it { expect(subject.school_and_university?).to be(false) }
+      end
+    end
+
+    describe "#online?" do
+      context "when the event is marked online and not virtual" do
+        let(:event) { build(:event_api, is_online: true, is_virtual: false) }
+
+        it { expect(subject).to be_online }
+      end
+
+      context "when the event is marked as virtual and not online" do
+        let(:event) { build(:event_api, is_online: false, is_virtual: true) }
+
+        it { expect(subject).to be_online }
+      end
+
+      context "when the event has a location" do
+        let(:event) { build(:event_api) }
+
+        it { expect(subject).to be_in_person }
+      end
+
+      context "when the event has no location" do
+        let(:event) { build(:event_api, :no_location) }
+
+        it { expect(subject).not_to be_in_person }
+      end
+    end
+
+    describe "#in_person?" do
+      context "when the event is a school and university event" do
+        let(:event) { build(:event_api, :virtual) }
+
+        it { expect(subject).to be_online }
```

Comment:
Not sure this test makes sense? should it not construct an event that's in person then check `be_in_person`?